### ISSUE TITLE
Add a linter to check indentations

### ISSUE THREAD

**Issue by yu-iskw** [r-lib/lintr#101]
Adding a new linter for validating indentations would be nice.

**Comment by jimhester** [comment]
I am not sure how to define what a valid indentation is.

```r
# Checking for indentation of 2
# valid
fun <- function(x) {
  x + 1
}

# invalid
fun <- function(x) {
 x + 1
}

# invalid
fun <- function(x) {
    x + 1
}

# but what about when aligning arguments, both the following should be valid?
fun <- function(x = 1,
                y = 2) {
  x + y
}

fun <- function(
  x = 1,
  y = 2) {
  x + y
}

# What is the proper indentation for anonymous functions?
# choice 1, aligned lapply
lapply(1:10, function(x) {
  x + 1
})

# choice 2, aligned to function
lapply(1:10, function(x) {
               x + 1
             }
)

# choice 3, function indented
lapply(1:10,
  function(x) {
    x + 1
  })
```

bottom line is I am unsure how to define a valid indentation style, and if we do have a valid definition, the proper algorithm to detect it.

**Comment by yu-iskw** [comment]
Yeah, it is quite difficult to define a valid indentation style. `formatR` package would help us with the problem, just idea.
http://yihui.name/formatR/

**Comment by MichaelChirico** [comment]
Closing as a duplicate of #63.
